scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -0,0 +1,35 @@
+use crate::compiler::plugin::proc_macro::PROC_MACRO_BUILD_PROFILE;
+use crate::core::{Config, Package};
+use crate::flock::Filesystem;
+use libloading::library_filename;
+
+/// This trait is used to define the shared library path for a package.
+pub trait SharedLibraryProvider {
+    /// Location of Cargo `target` directory.
+    fn target_path(&self, config: &Config) -> Filesystem;
+    /// Location of the shared library for the package.
+    fn shared_lib_path(&self, config: &Config) -> Filesystem;
+}
+
+impl SharedLibraryProvider for Package {
+    fn target_path(&self, config: &Config) -> Filesystem {
+        let ident = format!("{}-{}", self.id.name, self.id.source_id.ident());
+        // Defines the Cargo target directory in cache, as:
+        // `/(..)/SCARB_CACHE/plugins/proc_macro/<package_name>-<source_id_ident>/target/`
+        config
+            .dirs()
+            .procedural_macros_dir()
+            .into_child(ident)
+            .into_child("target")
+    }
+
+    fn shared_lib_path(&self, config: &Config) -> Filesystem {
+        let lib_name = library_filename(self.id.name.to_string());
+        let lib_name = lib_name.to_string_lossy().to_string();
```

Comment:
I think it'd be better to panic here, so that any non-UTF-8 shenanigans that the OS could try to sneak would trigger some kind of error, instead of giving a cryptic outcome that the library could not be found.